package.json

```diff
@@ -67,31 +68,32 @@
     "markdown-it": "13.0.1",
     "node-fetch": "^3.3.2",
     "recheck": "^4.4.5",
+    "rollup": "^3.27.0",
     "semantic-release": "^21.0.7",
     "titleize": "^3.0.0",
     "ts-expect": "^1.3.0",
     "ts-node": "^10.9.1",
-    "tsup": "^7.1.0",
     "typescript": "5.0.4",
-    "uglify-js": "^3.17.4",
-    "vuln-regex-detector": "^1.3.0"
+    "uglify-js": "^3.17.4"
   },
   "scripts": {
     "test": "cross-env NODE_OPTIONS=--loader=ts-node/esm jasmine --config=jasmine.json",
-    "test:all": "npm test && npm run test:lint",
+    "test:all": "npm test && npm run test:umd && npm run test:types && npm run test:lint",
     "test:unit": "npm test -- test/unit/**/*-spec.js",
     "test:specs": "npm test -- test/specs/**/*-spec.js",
     "test:lint": "eslint .",
     "test:redos": "node test/recheck.js > vuln.js",
+    "test:types": "tsc --project tsconfig-type-test.json",
+    "test:umd": "node test/umd-test.js",
     "test:update": "node test/update-specs.js",
     "rules": "node test/rules.js",
     "bench": "npm run build && node test/bench.js",
     "lint": "eslint --fix .",
-    "type-check": "tsc",
```

Comment:
Maybe I'm missing it, but where do we run `tsc` using the root `tsconfig.json`? Did you verify that a typo will fail the build?

Reply:
Yes rollup handles running tsc and will fail to build on typescript error